department-of-veterans-affairs/caseflow
# APPEALS-23363: Case Timeline Entry Details: Editing Designation for AMA appeals during Decision Workflow

Resolves #{https://jira.devops.va.gov/browse/APPEALS-23363}

# Description
Creates IssueUpdateTask object to display in CaseTimeline for AMA Appeals during the add decision workflow. IssueUpdateTask should include the following:
1. IssueUpdateTask header
2. Date issue edited
3. Name of person who made changes (completed by)
4. Issue for which special designation were changed
5. Original and Current special designations

## Acceptance Criteria

## Testing Plan
1. Switch user to BVAAABSHIRE
6. Go to Queue and select an AMA Appeal (e.g. Bob SmithMayer)
7. Click on the Task Action Dropdown menu and select "Ready for Dispatch"
8. If special issues page comes up (awaiting bugfix) click on "No Special issues" and hit "Continue"
9. On Add Decision page, click "Add Decision" for each issue and change the MST/PACT statuses
10. Click Continue and on fill out the first three fields of the Decision Evaluation Page
11. Click Continue and go back to Queue. Go to "Completed" appeals tab and click on Bob SmithMayer
12. Note that an IssueUpdateTask has been created for each decision and expand instructions to note that special issue designation changes are noted

# Frontend
1. Modified TaskRows.jsx to include conditional logic that removes the "Assigned To" detail display for IssueUpdateTasks

<img width="773" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/104021955/6e5eb710-ab97-4c41-8225-cab9c17f3240">

# Backend
1. Added a create_issue_update method to the case_reviews_controller.rb
2. Modified update_request_issues_for_mst_pact to accept an appeal as a parameter and added conditional logic to make mst/pact updates only when difference exist

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:
- [X ] Other - Manual validation

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [X ] No new code climate issues added
